### ISSUE TITLE
PCCS Docker build fails

### ISSUE THREAD

**Issue by pbeza** [intel/SGXDataCenterAttestationPrimitives#381]
I'm not able to build PCCS Docker image:

```
ubuntu@VM-0-6-ubuntu:/tmp$ git clone --recurse-submodules https://github.com/intel/SGXDataCenterAttestationPrimitives.git -b DCAP_1.20 --depth 1

ubuntu@VM-0-6-ubuntu:/tmp$ cd SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/container

ubuntu@VM-0-6-ubuntu:/tmp/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/container$ docker build -t pccs:my_tag .
[+] Building 107.1s (11/14)                                                                                                                       docker:default
 => [internal] load .dockerignore                                                                                                                           0.0s
 => => transferring context: 66B                                                                                                                            0.0s
 => [internal] load build definition from Dockerfile                                                                                                        0.0s  => => transferring dockerfile: 2.13kB                                                                                                                      0.0s
 => [internal] load metadata for docker.io/library/ubuntu:23.04                                                                                             2.4s 
 => [builder 1/7] FROM docker.io/library/ubuntu:23.04@sha256:5a828e28de105c3d7821c4442f0f5d1c52dc16acf4999d5f31a3bc0f03f06edd                               2.6s
 => => resolve docker.io/library/ubuntu:23.04@sha256:5a828e28de105c3d7821c4442f0f5d1c52dc16acf4999d5f31a3bc0f03f06edd                                       0.0s 
 => => sha256:ea1285dffce8a938ef356908d1be741da594310c8dced79b870d66808cb12b0f 424B / 424B                                                                  0.0s
 => => sha256:f4cdeba72b994748f5eb1f525a70a9cc553b66037ec37e23645fbf3f0f5c160d 2.30kB / 2.30kB                                                              0.0s
 => => sha256:6360b371721185fefbbad6763ab745900f1b2f7714570234473232dd575fc07f 26.89MB / 26.89MB                                                            1.8s 
 => => sha256:5a828e28de105c3d7821c4442f0f5d1c52dc16acf4999d5f31a3bc0f03f06edd 1.13kB / 1.13kB                                                              0.0s
 => => extracting sha256:6360b371721185fefbbad6763ab745900f1b2f7714570234473232dd575fc07f                                                                   0.7s 
 => [stage-1 2/5] RUN useradd -M -U -r pccs -s /bin/false                                                                                                   0.4s
 => [builder 2/7] RUN DEBIAN_FRONTEND=noninteractive     apt-get update -yq     && apt-get upgrade -yq     && apt-get install -yq --no-install-recommends  43.6s 
 => [builder 3/7] RUN git clone --recurse-submodules https://github.com/intel/SGXDataCenterAttestationPrimitives.git # -b DCAP_1.20 --depth 1              23.0s 
 => [builder 4/7] WORKDIR /SGXDataCenterAttestationPrimitives/tools/PCKCertSelection/                                                                       0.0s 
 => [builder 5/7] RUN make     && mkdir -p ../../QuoteGeneration/pccs/lib     && cp ./out/libPCKCertSelection.so ../../QuoteGeneration/pccs/lib/     && m  26.4s 
 => [builder 6/7] WORKDIR /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/                                                                         0.1s 
 => ERROR [builder 7/7] RUN npm config set proxy $http_proxy     && npm config set https-proxy $https_proxy     && npm config set engine-strict true     &  9.0s 
------                                                                                                                                                           
 > [builder 7/7] RUN npm config set proxy $http_proxy     && npm config set https-proxy $https_proxy     && npm config set engine-strict true     && npm install:
4.108 npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs                                                               
8.875 npm ERR! code 1                                                                                                                                            
8.875 npm ERR! path /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi                                                               
8.876 npm ERR! command failed                                                                                                                                    
8.876 npm ERR! command sh -c node-gyp-build                                                                                                                      
8.876 npm ERR! make: Entering directory '/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi/build'                                   
8.876 npm ERR!   CC(target) Release/obj.target/nothing/../node-addon-api/nothing.o                                                                               
8.876 npm ERR!   AR(target) Release/obj.target/../node-addon-api/nothing.a                                                                                       
8.876 npm ERR!   COPY Release/nothing.a                                                                                                                          
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/prep_cif.o                                                                                    8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/types.o      
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/raw_api.o                                                                                     
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/java_raw_api.o                                                                                
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/closures.o
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/x86/ffiw64.o
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/x86/ffi64.o
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/x86/unix64.o
8.876 npm ERR!   CC(target) Release/obj.target/ffi/deps/libffi/src/x86/win64.o
8.876 npm ERR!   AR(target) Release/obj.target/deps/libffi/libffi.a
8.876 npm ERR!   COPY Release/libffi.a
8.876 npm ERR!   CXX(target) Release/obj.target/ffi_bindings/src/ffi.o
8.876 npm ERR! make: Leaving directory '/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi/build'
8.877 npm ERR! gyp info it worked if it ends with ok
8.877 npm ERR! gyp info using node-gyp@8.4.1
8.877 npm ERR! gyp info using node@20.12.0 | linux | x64
8.877 npm ERR! gyp info find Python using Python version 3.11.4 found at "/usr/bin/python3"
8.877 npm ERR! gyp http GET https://nodejs.org/download/release/v20.12.0/node-v20.12.0-headers.tar.gz
8.877 npm ERR! gyp http 200 https://nodejs.org/download/release/v20.12.0/node-v20.12.0-headers.tar.gz
8.877 npm ERR! gyp http GET https://nodejs.org/download/release/v20.12.0/SHASUMS256.txt
8.877 npm ERR! gyp http 200 https://nodejs.org/download/release/v20.12.0/SHASUMS256.txt
8.877 npm ERR! gyp info spawn /usr/bin/python3
8.877 npm ERR! gyp info spawn args [
8.877 npm ERR! gyp info spawn args   '/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-gyp/gyp/gyp_main.py',
8.877 npm ERR! gyp info spawn args   'binding.gyp',
8.877 npm ERR! gyp info spawn args   '-f',
8.877 npm ERR! gyp info spawn args   'make',
8.877 npm ERR! gyp info spawn args   '-I',
8.877 npm ERR! gyp info spawn args   '/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi/build/config.gypi',
8.877 npm ERR! gyp info spawn args   '-I',
8.877 npm ERR! gyp info spawn args   '/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-gyp/addon.gypi',
8.877 npm ERR! gyp info spawn args   '-I',
8.877 npm ERR! gyp info spawn args   '/root/.cache/node-gyp/20.12.0/include/node/common.gypi',
8.877 npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
8.877 npm ERR! gyp info spawn args   '-Dvisibility=default',
8.877 npm ERR! gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/20.12.0',
8.877 npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-gyp',
8.877 npm ERR! gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/20.12.0/<(target_arch)/node.lib',
8.877 npm ERR! gyp info spawn args   '-Dmodule_root_dir=/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi',
8.877 npm ERR! gyp info spawn args   '-Dnode_engine=v8',
8.877 npm ERR! gyp info spawn args   '--depth=.',
8.877 npm ERR! gyp info spawn args   '--no-parallel',
8.877 npm ERR! gyp info spawn args   '--generator-output',
8.877 npm ERR! gyp info spawn args   'build',
8.877 npm ERR! gyp info spawn args   '-Goutput_dir=.'
8.877 npm ERR! gyp info spawn args ]
8.877 npm ERR! gyp info spawn make
8.877 npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
8.877 npm ERR! ../deps/libffi/src/java_raw_api.c: In function 'ffi_java_raw_call':
8.877 npm ERR! ../deps/libffi/src/java_raw_api.c:317:3: warning: 'ffi_java_raw_to_ptrarray' is deprecated [-Wdeprecated-declarations]
8.877 npm ERR!   317 |   ffi_java_raw_to_ptrarray (cif, raw, avalue);
8.878 npm ERR!       |   ^~~~~~~~~~~~~~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:76:1: note: declared here
8.878 npm ERR!    76 | ffi_java_raw_to_ptrarray (ffi_cif *cif, ffi_java_raw *raw, void **args)
8.878 npm ERR!       | ^~~~~~~~~~~~~~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c: In function 'ffi_java_translate_args':
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:328:3: warning: 'ffi_java_raw_size' is deprecated [-Wdeprecated-declarations]
8.878 npm ERR!   328 |   ffi_java_raw *raw = (ffi_java_raw*)alloca (ffi_java_raw_size (cif));
8.878 npm ERR!       |   ^~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:45:1: note: declared here
8.878 npm ERR!    45 | ffi_java_raw_size (ffi_cif *cif)
8.878 npm ERR!       | ^~~~~~~~~~~~~~~~~ 
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:331:3: warning: 'ffi_java_ptrarray_to_raw' is deprecated [-Wdeprecated-declarations]
8.878 npm ERR!   331 |   ffi_java_ptrarray_to_raw (cif, avalue, raw);
8.878 npm ERR!       |   ^~~~~~~~~~~~~~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:157:1: note: declared here
8.878 npm ERR!   157 | ffi_java_ptrarray_to_raw (ffi_cif *cif, void **args, ffi_java_raw *raw)
8.878 npm ERR!       | ^~~~~~~~~~~~~~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c: In function 'ffi_prep_java_raw_closure':
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:369:3: warning: 'ffi_prep_java_raw_closure_loc' is deprecated [-Wdeprecated-declarations]
8.878 npm ERR!   369 |   return ffi_prep_java_raw_closure_loc (cl, cif, fun, user_data, cl);
8.878 npm ERR!       |   ^~~~~~
8.878 npm ERR! ../deps/libffi/src/java_raw_api.c:337:1: note: declared here
8.878 npm ERR!   337 | ffi_prep_java_raw_closure_loc (ffi_java_raw_closure* cl,
8.878 npm ERR!       | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8.878 npm ERR! ../deps/libffi/src/closures.c: In function 'allocate_space':
8.878 npm ERR! ../deps/libffi/src/closures.c:748:29: warning: comparison of integer expressions of different signedness: 'off_t' {aka 'long int'} and 'size_t' {a
ka 'long unsigned int'} [-Wsign-compare] 
8.878 npm ERR!   748 |       off_t to_write = (len < page_size) ? len : page_size;
8.878 npm ERR!       |                             ^
8.878 npm ERR! ../deps/libffi/src/closures.c:748:44: warning: operand of '?:' changes signedness from 'off_t' {aka 'long int'} to 'size_t' {aka 'long unsigned in
t'} due to unsignedness of other operand [-Wsign-compare]
8.878 npm ERR!   748 |       off_t to_write = (len < page_size) ? len : page_size;
8.878 npm ERR!       |                                            ^~~
8.879 npm ERR! ../deps/libffi/src/x86/ffi64.c: In function 'classify_argument':
8.879 npm ERR! ../deps/libffi/src/x86/ffi64.c:200:25: warning: suggest braces around empty body in an 'else' statement [-Wempty-body]
8.879 npm ERR!   200 |           FFI_ASSERT (0);
8.879 npm ERR!       |                         ^
8.879 npm ERR! ../deps/libffi/src/x86/ffi64.c:175:7: warning: this statement may fall through [-Wimplicit-fallthrough=]
8.879 npm ERR!   175 |       {
8.879 npm ERR!       |       ^
8.879 npm ERR! ../deps/libffi/src/x86/ffi64.c:202:5: note: here
8.879 npm ERR!   202 |     case FFI_TYPE_FLOAT:
8.879 npm ERR!       |     ^~~~
8.879 npm ERR! In file included from ../src/ffi.cc:5:
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/get-uv-event-loop-napi-h/include/get-uv-event-loop-napi.h: In function 'uv_l
oop_t* get_uv_event_loop(napi_env)':
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/get-uv-event-loop-napi-h/include/get-uv-event-loop-napi.h:26:30: error: inva
lid conversion from 'napi_status (*)(node_api_nogc_env, uv_loop_s**)' {aka 'napi_status (*)(const napi_env__*, uv_loop_s**)'} to 'get_uv_event_loop_fn' {aka 'nap
i_status (*)(napi_env__*, uv_loop_s**)'} [-fpermissive]
8.879 npm ERR!    26 |   napi_get_uv_event_loop__ = &napi_get_uv_event_loop;
8.879 npm ERR!       |                              ^~~~~~~~~~~~~~~~~~~~~~~
8.879 npm ERR!       |                              |
8.879 npm ERR!       |                              napi_status (*)(node_api_nogc_env, uv_loop_s**) {aka napi_status (*)(const napi_env__*, uv_loop_s**)}
8.879 npm ERR! In file included from /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi.h:2725,
8.879 npm ERR!                  from ../src/ffi.h:2,
8.879 npm ERR!                  from ../src/ffi.cc:3:
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h: In instantiation of 'napi_status Napi::details::A
ttachData(napi_env, napi_value, FreeType*, napi_finalize, void*) [with FreeType = CallbackData<Napi::Value (*)(const Napi::CallbackInfo&), Napi::Value>; napi_env
 = napi_env__*; napi_value = napi_value__*; napi_finalize = void (*)(napi_env__*, void*, void*)]':
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:1965:39:   required from 'napi_status Napi::Create
Function(napi_env, const char*, napi_callback, CbData*, napi_value__**) [with CbData = details::CallbackData<Value (*)(const CallbackInfo&), Value>; napi_env = n
api_env__*; napi_callback = napi_value__* (*)(napi_env__*, napi_callback_info__*); napi_value = napi_value__*]'
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:2021:38:   required from 'static Napi::Function Na
pi::Function::New(napi_env, Callable, const char*, void*) [with Callable = Napi::Value (*)(const Napi::CallbackInfo&); napi_env = napi_env__*]'
8.879 npm ERR! ../src/ffi.cc:58:41:   required from here
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:63:47: error: invalid conversion from 'napi_finali
ze' {aka 'void (*)(napi_env__*, void*, void*)'} to 'node_api_nogc_finalize' {aka 'void (*)(const napi_env__*, void*, void*)'} [-fpermissive]
8.879 npm ERR!    63 |   status = napi_add_finalizer(env, obj, data, finalizer, hint, nullptr);
8.879 npm ERR!       |                                               ^~~~~~~~~
8.879 npm ERR!       |                                               |
8.879 npm ERR!       |                                               napi_finalize {aka void (*)(napi_env__*, void*, void*)}
8.879 npm ERR! In file included from /root/.cache/node-gyp/20.12.0/include/node/node_api.h:12,
8.879 npm ERR!                  from /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi.h:4:
8.879 npm ERR! /root/.cache/node-gyp/20.12.0/include/node/js_native_api.h:523:43: note:   initializing argument 4 of 'napi_status napi_add_finalizer(napi_env, na
pi_value, void*, node_api_nogc_finalize, void*, napi_ref__**)'
8.879 npm ERR!   523 |                    node_api_nogc_finalize finalize_cb,
8.879 npm ERR!       |                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
8.879 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h: In instantiation of 'napi_status Napi::details::A
ttachData(napi_env, napi_value, FreeType*, napi_finalize, void*) [with FreeType = CallbackData<void (*)(const Napi::CallbackInfo&), void>; napi_env = napi_env__*
; napi_value = napi_value__*; napi_finalize = void (*)(napi_env__*, void*, void*)]':
8.880 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:1965:39:   required from 'napi_status Napi::Create
Function(napi_env, const char*, napi_callback, CbData*, napi_value__**) [with CbData = details::CallbackData<void (*)(const CallbackInfo&), void>; napi_env = nap
i_env__*; napi_callback = napi_value__* (*)(napi_env__*, napi_callback_info__*); napi_value = napi_value__*]'
8.880 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:2021:38:   required from 'static Napi::Function Na
pi::Function::New(napi_env, Callable, const char*, void*) [with Callable = void (*)(const Napi::CallbackInfo&); napi_env = napi_env__*]'
8.880 npm ERR! ../src/ffi.cc:60:37:   required from here
8.880 npm ERR! /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-addon-api/napi-inl.h:63:47: error: invalid conversion from 'napi_finali
ze' {aka 'void (*)(napi_env__*, void*, void*)'} to 'node_api_nogc_finalize' {aka 'void (*)(const napi_env__*, void*, void*)'} [-fpermissive]
8.880 npm ERR!    63 |   status = napi_add_finalizer(env, obj, data, finalizer, hint, nullptr);
8.880 npm ERR!       |                                               ^~~~~~~~~
8.880 npm ERR!       |                                               |
8.880 npm ERR!       |                                               napi_finalize {aka void (*)(napi_env__*, void*, void*)}
8.880 npm ERR! /root/.cache/node-gyp/20.12.0/include/node/js_native_api.h:523:43: note:   initializing argument 4 of 'napi_status napi_add_finalizer(napi_env, na
pi_value, void*, node_api_nogc_finalize, void*, napi_ref__**)'
8.880 npm ERR!   523 |                    node_api_nogc_finalize finalize_cb,
8.880 npm ERR!       |                    ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
8.880 npm ERR! make: *** [ffi_bindings.target.mk:121: Release/obj.target/ffi_bindings/src/ffi.o] Error 1
8.880 npm ERR! gyp ERR! build error 
8.880 npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
8.880 npm ERR! gyp ERR! stack     at ChildProcess.onExit (/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-gyp/lib/build.js:194:23)
8.880 npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:518:28)
8.880 npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
8.880 npm ERR! gyp ERR! System Linux 6.5.0-1011-oem
8.880 npm ERR! gyp ERR! command "/usr/bin/node" "/SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
8.880 npm ERR! gyp ERR! cwd /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/node_modules/ffi-napi
8.880 npm ERR! gyp ERR! node -v v20.12.0 
8.880 npm ERR! gyp ERR! node-gyp -v v8.4.1
8.880 npm ERR! gyp ERR! not ok
8.881 
8.882 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-03-27T16_26_14_377Z-debug-0.log
------
Dockerfile:38
--------------------
  37 |     WORKDIR /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/
  38 | >>> RUN npm config set proxy $http_proxy \
  39 | >>>     && npm config set https-proxy $https_proxy \
  40 | >>>     && npm config set engine-strict true \
  41 | >>>     && npm install
  42 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c npm config set proxy $http_proxy     && npm config set https-proxy $https_proxy     && npm config set engine-strict t
rue     && npm install" did not complete successfully: exit code: 1
```

Same issue if I use:
```
git clone --recurse-submodules https://github.com/intel/SGXDataCenterAttestationPrimitives.git
```
instead of:
```
git clone --recurse-submodules https://github.com/intel/SGXDataCenterAttestationPrimitives.git -b DCAP_1.20 --depth 1
```

Could the PCCS code be messed up, or am I missing something? It was running fine for me before, so maybe something changed with the PCCS dependencies? Any idea @llly?

**Comment by lingyuj** [comment]
It looks like the latest node.js minor version update(20.12) broke the node-ffi module. Can you please use node.js 20.11 to have a retry?

**Comment by pbeza** [comment]
It worked!

```
diff --git a/QuoteGeneration/pccs/container/Dockerfile b/QuoteGeneration/pccs/container/Dockerfile
index 3a70c44..37a7c1f 100644
--- a/QuoteGeneration/pccs/container/Dockerfile
+++ b/QuoteGeneration/pccs/container/Dockerfile
@@ -19,7 +19,7 @@ RUN DEBIAN_FRONTEND=noninteractive \
     && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /usr/share/keyrings/nodesource.gpg \
     && echo "deb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODE_MAJOR}.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
     && apt-get update -yq \
-    && apt-get install -yq --no-install-recommends nodejs \
+    && apt-get install -yq --no-install-recommends nodejs=20.11.1-1nodesource1 \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
```

Could you please fix it upstream? Additionally, why not use the latest `nodejs` version, which has a major number of `21`, instead of `20`? This repository is lagging behind the latest versions of its dependencies. :(

**Comment by llly** [comment]
Fixed in 1.21 release
